org/qmckl_jastrow.org

```diff
@@ -1304,42 +1303,54 @@ integer function qmckl_compute_asymp_jasb_f(context, bord_num, bord_vector, resc
 end function qmckl_compute_asymp_jasb_f
     #+end_src
 
-    #+CALL: generate_c_header(table=qmckl_asymp_jasb_args,rettyp=get_value("CRetType"),fname=get_value("Name"))
-
-    #+RESULTS:
-    #+BEGIN_src c :tangle (eval h_func) :comments org
-    qmckl_exit_code qmckl_compute_asymp_jasb (
+#+begin_src c :comments org :tangle (eval c) :noweb yes
+qmckl_exit_code qmckl_compute_asymp_jasb (
 	  const qmckl_context context,
 	  const int64_t bord_num,
 	  const double* bord_vector,
 	  const double rescale_factor_kappa_ee,
-	  double* const asymp_jasb );
-    #+END_src
+	  double* const asymp_jasb ) {
 
+    double    kappa_inv, x, asym_one;
 
-    #+CALL: generate_c_interface(table=qmckl_asymp_jasb_args,rettyp=get_value("CRetType"),fname=get_value("Name"))
+    kappa_inv = 1.0 / rescale_factor_kappa_ee;
 
-    #+RESULTS:
-    #+BEGIN_src f90 :tangle (eval f) :comments org :exports none
-    integer(c_int32_t) function qmckl_compute_asymp_jasb &
-	(context, bord_num, bord_vector, rescale_factor_kappa_ee, asymp_jasb) &
-	bind(C) result(info)
+    if (context == QMCKL_NULL_CONTEXT){
+      return QMCKL_INVALID_CONTEXT;
+    }
 
-      use, intrinsic :: iso_c_binding
-      implicit none
+    if (bord_num <= 0) {
+      return QMCKL_INVALID_ARG_2;
+    }
```

Comment:
Nice that you put the checks for the arguments !

Reply:
Thanks for the credits, but the checks were already in place, so kudos to you!

> Would you like to merge this or keep pushing to this PR (with a name change) for other functions also ?

For me it's the same. If it is less annoying for you, just use this one and I'll add commit to this.

Reply:
I'll merge this for now.
Maybe for the next time it would be better if you send one PR for the whole Jastrow module.